R/csv.R

```diff
@@ -248,6 +246,10 @@ pl$read_csv = function(
 
 
 check_is_link = function(path, reuse_downloaded, raise_error = FALSE) {
+  # do nothing let path fail on rust side
+  if (is.na(path)) {
+    return(NULL)
+  }
```

Comment:
Otherwise, `NA` goes through all the code about connections (without erroring), which is surprisingly slow. Better to have an early exit here